In [1]:
print('Hello World')

Hello World


In [ ]:
import pandas as pd

class ExcelMerger:
    def __init__(self, file_paths):
        self.file_paths = file_paths

    def merge_sheets(self):
        combined_data = pd.DataFrame()
        for file_path in self.file_paths:
            excel_file = pd.ExcelFile(file_path)
            for sheet_name in excel_file.sheet_names:
                df = excel_file.parse(sheet_name)
                combined_data = pd.concat([combined_data, df])
        return combined_data

    def save_to_excel(self, combined_data, output_file):
        combined_data.to_excel(output_file, index=False)

# Example usage
file_paths = ['file1.xlsx', 'file2.xlsx']  # List of file paths
merger = ExcelMerger(file_paths)
combined_data = merger.merge_sheets()
merger.save_to_excel(combined_data, 'combined_file.xlsx')


In [6]:
## OOP 1
import pandas as pd
class Merger:
    def __init__(self, files):
        self.files =files
        
    def iterate(self):
        combined = pd.DataFrame()
        for file in self.files: # Iterating through the files
            print('i')
            excel = pd.ExcelFile(file) # Opening each file
            for sheet in excel.sheet_names: # Opening each sheets of the file
                df = excel.parse(sheet)
                combined = pd.concat([combined, df])
        return(combined)


files = ['Barangka.xlsx', 'Nangka.xlsx', 'Parang.xlsx']
merger = Merger(files)
comb_data =merger.iterate()
print(comb_data)
        

i
i
i
        Unnamed: 0 Unnamed: 1 Unnamed: 2 Unnamed: 3  \
0   Respondent No.   Barangay     Gender        Age   
1                1   Barangka     Female      18-24   
2                2   Barangka     Female      18-24   
3                3   Barangka     Female      25-34   
4                4   Barangka     Female      25-34   
..             ...        ...        ...        ...   
20              20     Tumana          F      35-44   
21              21     Tumana          M      55-64   
22              22     Tumana          M      35-44   
23              23     Tumana          F      55-64   
24              24     Tumana          M      55-64   

                          Unnamed: 4              Unnamed: 5     Unnamed: 6  \
0   Average Household Monthly Income  Educational Attainment     Employment   
1          P 25, 001.00 - 50, 000.00        College Graduate       Employed   
2          P 25, 001.00 - 50, 000.00        College Graduate       Employed   
3          P 25, 

In [27]:
comb_data.shape

(474, 62)

Based on the paper, there are total of 383 respondents. Thus, 474 rows is absurd. One way to address this is by checking if there is duplicates. If yes, then drop it.

In [34]:
dupl = comb_data.duplicated()
print(comb_data[dupl])

        Unnamed: 0 Unnamed: 1 Unnamed: 2 Unnamed: 3  \
0   Respondent No.   Barangay     Gender        Age   
0   Respondent No.   Barangay     Gender        Age   
0   Respondent No.   Barangay     Gender        Age   
0   Respondent No.   Barangay     Gender        Age   
27             NaN        NaN        NaN        NaN   
..             ...        ...        ...        ...   
0   Respondent No.   Barangay     Gender        Age   
0   Respondent No.   Barangay     Gender        Age   
0   Respondent No.   Barangay     Gender        Age   
0   Respondent No.   Barangay     Gender        Age   
0   Respondent No.   Barangay     Gender        Age   

                          Unnamed: 4              Unnamed: 5  Unnamed: 6  \
0   Average Household Monthly Income  Educational Attainment  Employment   
0   Average Household Monthly Income  Educational Attainment  Employment   
0   Average Household Monthly Income  Educational Attainment  Employment   
0   Average Household Monthly Incom

In [38]:
comb_data_1 = comb_data.drop_duplicates()
comb_data_1.shape

(388, 62)

Save the new dataframe into a new excel file

In [40]:
comb_data.to_excel('combined_data.xlsx')

OOP2 Address:
1. Repeated column names
2. Duplicates = check
3. 'Nangka.xlsx' dirty file = check
3. Save it to excel right away


In [10]:
## OOP2
import pandas as pd
class Merger1:
    def __init__(self, files):
        self.files = files

    def iterate(self):
        comb = pd.DataFrame
        for file in self.files:
            print('Analyzing')
            # opening the file
            excel = pd.ExcelFile(file)
            # going through the sheets of the file using '.sheet_names'
            for sheet in excel.sheet_names:
                # parse the sheet to the excel
                df = excel.parse(sheet)
                first = df.shape
                # removing null values using thresh = atleast 5 null values in a row
                df = df.dropna(thresh=5)  
                second = df.shape
                if first == second:
                    print(df.shape)
                else:
                    print('Inconsistent shape...\nHandled')
                # combining the data together
                comb = pd.concat([comb, df])
        comb_1 = comb.drop_duplicates()
        print(comb_1.shape)
        return(comb_1)
    
    def save(self, combined_data, output_file):
        combined_data.to_excel(output_file, index=False)

     
            
files = ['Barangka.xlsx', 'Nangka.xlsx', 'Parang.xlsx']
merger = Merger1(files)
comb_data = merger.iterate()
merger.save(comb_data, 'combined_data(1).xlsx')
        

In [18]:
## OOP 1
import pandas as pd
class Merger:
    def __init__(self, files):
        self.files =files
        
    def iterate(self):
        combined = pd.DataFrame()
        for file in self.files: # Iterating through the files
            print('i')
            excel = pd.ExcelFile(file) # Opening each file
            for sheet in excel.sheet_names: # Opening each sheets of the file
                df = excel.parse(sheet)
                first = df.shape
                df = df.dropna(thresh=5) 
                second = df.shape 
                if first == second:
                    print(second)
                else:
                    print(first, '-->', second)               
                
                combined = pd.concat([combined, df])
        comb_1 = combined.drop_duplicates()
        print('Total index of the table is', comb_1.shape)
        return(comb_1)
    
    def save(self, combined_data, output_file):
        combined_data.to_excel(output_file, index=False)
        return('File is saved')

files = ['Barangka.xlsx', 'Nangka.xlsx', 'Parang.xlsx']
merger = Merger(files)
comb_data =merger.iterate()
merger.save(comb_data, 'combined_data(1).xlsx')

        

i
(25, 61)
(25, 61)
(25, 61)
(25, 61)
(25, 61)
i
(99, 61) --> (25, 61)
(25, 61)
(25, 61)
(25, 61)
(25, 61)
i
(25, 61)
(25, 61)
(25, 61)
(25, 61)
(25, 61)
(25, 61)
Total index of the table is (386, 62)


'File is saved'

Data Manipulation

In [20]:
df = pd.read_csv('combined_data(1).csv')

In [21]:
df.columns

Index(['Respondent No.', 'Barangay', 'Gender', 'Age', 'Ave_Monthly_Inc',
       'Educ_Attain', 'Employment', 'Years_of_Stay', 'F1', 'F2', 'E1', 'E2',
       'O1', 'O2', 'OR1', 'OR2', 'OR3', 'I1', 'I2', 'I3', 'I4', 'I5', 'SF1',
       'SF2', 'SF3', 'SF4', 'PR1', 'PR2', 'PR3', 'PR4', 'SN1', 'SN2', 'SN3',
       'SIC1', 'SIC2', 'SIC3', 'SIC4', 'SIC5', 'SIC6', 'US1', 'US2', 'US3',
       'US4', 'IIR1', 'IIR2', 'IIR3', 'IIR4', 'IIR5', 'IIR6', 'IIR7', 'IIR8',
       'IIR9', 'EED1', 'EED2', 'EED3', 'EED4', 'EED5', 'EED6', 'EED7', 'ERB1',
       'ERB2'],
      dtype='object')

In [22]:
df['Gender'].unique()

array(['Female', 'Male'], dtype=object)

In [23]:
df['Age'].unique()

array(['18-24', '25-34', '35-44', '45-54', '55-64', 'Above 64',
       'above 64', '64+'], dtype=object)

In [24]:
df['Age'] = df['Age'].replace('above 64', '64+')
df['Age'] = df['Age'].replace('Above 64', '64+')

In [25]:
df['Age'].unique()

array(['18-24', '25-34', '35-44', '45-54', '55-64', '64+'], dtype=object)

In [29]:
df['Ave_Monthly_Inc'].unique()

array(['P 25, 001.00 - 50, 000.00', 'P 10, 001.00 - 25, 000.00',
       'P 10, 000.00 and below', 'P 50, 001.00 and above', 'Below 10k',
       '10k-25k', '25k-50k', '50k above', '10,000 below', '25,001-50,000'],
      dtype=object)

In [32]:
df['Ave_Monthly_Inc'] = df['Ave_Monthly_Inc'].replace('P 25, 001.00 - 50, 000.00', '25k-50k')
df['Ave_Monthly_Inc'] = df['Ave_Monthly_Inc'].replace('25,001-50,000', '25k-50k')
df['Ave_Monthly_Inc'] = df['Ave_Monthly_Inc'].replace('10,000 below', 'Below 10k')
df['Ave_Monthly_Inc'] = df['Ave_Monthly_Inc'].replace('P 10, 001.00 - 25, 000.00', '10k-25k')
df['Ave_Monthly_Inc'] = df['Ave_Monthly_Inc'].replace('P 10, 000.00 and below', 'Below 10k')
df['Ave_Monthly_Inc'] = df['Ave_Monthly_Inc'].replace('P 50, 001.00 and above', '50k above')

In [33]:
df['Ave_Monthly_Inc'].unique()

array(['25k-50k', '10k-25k', 'Below 10k', '50k above'], dtype=object)

In [34]:
df['Educ_Attain'].unique()

array(['College Graduate', 'College Level', 'Vocational',
       'High School Graduate', 'HS GRAD', 'College Grad', 'Vocational ',
       'Senior Highschool Graduate'], dtype=object)

In [35]:
df['Educ_Attain'] = df['Educ_Attain'].replace('HS GRAD', 'High School Graduate')
df['Educ_Attain'] = df['Educ_Attain'].replace('College Grad', 'College Graduate')
df['Educ_Attain'] = df['Educ_Attain'].replace('Vocational ', 'Vocational')

In [36]:
df['Educ_Attain'].unique()

array(['College Graduate', 'College Level', 'Vocational',
       'High School Graduate', 'Senior Highschool Graduate'], dtype=object)

In [37]:
df['Employment'].unique()

array(['Employed', 'Self-employed', 'Unemployed', 'Self-Employed',
       'Self Employed', 'Unemployed '], dtype=object)

In [40]:
df['Employment'] = df['Employment'].replace(['Self Employed', 'Self-Employed'], 'Self-employed')
df['Employment'] = df['Employment'].replace('Unemployed ', 'Unemployed')

In [41]:
df['Employment'].unique()

array(['Employed', 'Self-employed', 'Unemployed'], dtype=object)

In [42]:
df['Years_of_Stay'].unique()

array(['20 years and above', '5-10 years', '5 years and below',
       '10-15 years', '15-20 years', '20-years above', '5yrs below',
       '21 years and above'], dtype=object)

In [43]:
df['Years_of_Stay'] = df['Years_of_Stay'].replace(['20-years above', '20 years and above'], '21 years and above')
df['Years_of_Stay'] = df['Years_of_Stay'].replace(['5yrs below'], '5 years and below')

In [44]:
df['Years_of_Stay'].unique()

array(['21 years and above', '5-10 years', '5 years and below',
       '10-15 years', '15-20 years'], dtype=object)

In [45]:
df.to_csv('combined_data(2).csv')